src/library_browser.js

```diff
@@ -908,7 +908,7 @@ var LibraryBrowser = {
         // if the destination directory does not yet exist, create it
         FS.mkdirTree(destinationDirectory);
 
-        FS.createDataFile( _file.substr(0, index), _file.substr(index + 1), new Uint8Array(http.response), true, true, false);
+        FS.createDataFile( _file.substr(0, index), _file.substr(index + 1), new Uint8Array(/** @type{ArrayBuffer}*/(http.response)), true, true, false);
```

Comment:
I'm curious, what does this ArrayBuffer annotation actually do? Does it make closure check all callers provide the proper type? And/or does it prevent a warning of some kind?

Reply:
By default Closure (rightfully) thinks that `xhr.response` can be a null, Object, string, Array, ArrayBuffer or some other types, but Uint8Array is only valid to be created from an ArrayBuffer. We pass `.responsetype = 'arraybuffer';` above, but Closure does not understand that.